### PR TITLE
[VOY-610] [DEV-ONLY] Update repo to match standards.

### PR DESCRIPTION
**Pull request by elijahr**
Ticket
---
https://styleseat.atlassian.net/browse/VOY-610

Description
---
Make this repository conform to our [checklist](https://styleseat.atlassian.net/wiki/spaces/EN/pages/1620115476/StyleSeat+Git+Repository+Standards). 

- [ ] Rename repo (name seems fine - discuss?)
- [x] Add pull request template
- [x] Configure protected branches (renamed `master` to `dev`, added `release`)
- [x] Assign ownership in spreadsheet (already done)
- [x] README (already exists)
- [x] JIRA integration (already configured)

Also:
- [x] Add lintball, GitHub workflow for linting, and pre-commit hook.
- [x] Reformat files with black
- [x] Test against Python 2.7, 3.8 (was Python 2.7, 3.7).

PR Checklist
---
- [n/a] Unit tests have been added and the test suite passes.
- [x] README and/or Confluence documentation has been updated if needed.

### CODE REVIEW

**Review by CaleHarter — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+*Place links to other Pull Requests here.*
+
+How to Test
+---
+*Instructions for testing, if needed.*
+
+PR Checklist
+---
+- [ ] Unit tests have been added and the test suite passes.
+- [ ] README and/or Confluence documentation has been updated, if needed.
```
Comment (amethystmarie) [inline .github/pull_request_template.md:24]: I think there should be a checklist item for bumping major or minor version as necessary and updating the referenced version in dependent repos.

File: setup.py [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
-      package_dir={'': 'src'},
-      install_requires=['boto3>1'],
-      zip_safe=True)
-
+setup(
+    name="ss_instrumentation",
+    version="2.0.0",
+    description="",
+    url="https://github.com/styleseat/ss-instrumentation",
+    author="Some Dude at StyleSeat",
```
Comment (amethystmarie) [inline setup.py]: Can we change the author to be something real instead of a joke?

**Review by amethystmarie — CHANGES_REQUESTED** [review]
We'll also need a corresponding PR in the styleseat repo to update our usage of ss-instrumentation to point to the new version.

**Comment by amethystmarie** [comment]
Can we automate the tagging when the version number changes?

**Comment by elijahr** [comment]
> Can we automate the tagging when the version number changes?

@amethystmarie interesting point. Cursory research leads me to https://github.com/laputansoft/github-tag-action. Looks like that will do what we want, if we provide the `tag` parameter, so that it doesn't automatically bump the version for us.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ss-instrumentation",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (elijahr) [inline package.json:3]: Note for self: why does this package have both a setup.py and a package.json?

Comment (elijahr) [reply package.json:3]: (becase npm is used for deploying this, via serverless)

**Review by CaleHarter — APPROVED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -49,14 +84,31 @@ workflows:
...
     jobs:
       - toxtest
+      - tag_release_hold:
+          type: approval
+          requires:
+            - toxtest
+          filters:
+            branches:
+              only:
+                - dev
```
Comment (amethystmarie) [inline .circleci/config.yml]: What do you think about not having a hold on tagging the release? Who would be pushing this button if we keep it?

**Review by amethystmarie — COMMENTED** [review]
the deploy_stage and deploy_prod don't really do anything, right?

**Comment by elijahr** [comment]
@amethystmarie 

> the deploy_stage and deploy_prod don't really do anything, right?

from what i can tell, they use serverless to deploy `flusher.py`, which runs every minute. presumably this service is still running.

**Review by amethystmarie — APPROVED** [review]

**Pull request merged**